### PR TITLE
Update usage.md

### PR DESCRIPTION
**Pull request by sofstam**
This PR updates the usage.md
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by sofstam** [comment]
@nf-core-bot fix linting

**Comment by sofstam** [comment]
@nf-core-bot fix linting

**Comment by sofstam** [comment]
@nf-core-bot fix linting

**Review by jfy133 — COMMENTED** [review]
Whelp, apparnetly I wrote everything and didn't press 'submit review' :sweat_smile:

File: docs/usage.md [hunk 1]
```diff
@@ -220,10 +220,14 @@ Complexity filtering is primarily a run-time optimisation step. It is not necess
...
+There is one option for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong)
+
+The tools offer different algorithms and parameters for removing low complexity reads and quality filtering. We therefore recommend reviewing the pipeline's [parameter documentation](https://nf-co.re/taxprofiler/parameters) and the documentation of the tools (see links above) to decide on optimal methods and parameters for your dataset.
 
 You can optionally save the FASTQ output of the run merging with the `--save_complexityfiltered_reads`. If running with `fastp`, complexity filtering happens inclusively within the earlier shortread preprocessing step. Therefore there will not be an independent pipeline step for complexity filtering, and no independent FASTQ file (i.e. `--save_complexityfiltered_reads` will be ignored) - your complexity filtered reads will also be in the `fastp/` folder in the same file(s) as the preprocessed read.
 
+**We do not any read preprocessing or complexity filtering if you are using ONTs Guppy toolkit for basecalling and post-processing.**
```
[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md:229]: I meant: 

**We do not recommend performing any read preprocessing or complexity filtering if you are already using ONTs Guppy toolkit for basecalling and post-processing.**

File: docs/usage.md [hunk 2]
```diff
@@ -475,6 +479,53 @@ We recommend adding the following line to your environment to limit this (typica
...
+wget https://software-ab.informatik.uni-tuebingen.de/download/megan6/megan-nucl-Feb2022.db.zip
+unzip megan-nucl-Feb2022.db
+malt-build -i path/to/fasta/files/*.{fna,fa} -s DNA -d index -t 8 -st 4 -a2t megan-nucl-Feb2022.db
+```
+
+#### Bracken
+You can follow Bracken [tutorial](https://ccb.jhu.edu/software/bracken/index.shtml?t=manual) to build a custom database. Alternatively, you can use one of the indexes that can be found [here](https://benlangmead.github.io/aws-indexes/k2).
+
+#### KrakenUniq
+For KrakenUniq, we recommend using one of the available databases [here](https://benlangmead.github.io/aws-indexes/k2)
```
[the remark this answers is not in the document]
Comment (jfy133) [reply docs/usage.md:514]: Ah sorry, we should say 'But if you wish to build your own, please see XYZ documentation' - apparenrtly I didn't finish the review :laughing:

[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md:514]: Ok :D

File: docs/usage.md [hunk 3]
```diff
@@ -220,10 +221,14 @@ Complexity filtering is primarily a run-time optimisation step. It is not necess
...
+There is one option for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong)
+
+The tools offer different algorithms and parameters for removing low complexity reads and quality filtering. We therefore recommend reviewing the pipeline's [parameter documentation](https://nf-co.re/taxprofiler/parameters) and the documentation of the tools (see links above) to decide on optimal methods and parameters for your dataset.
 
 You can optionally save the FASTQ output of the run merging with the `--save_complexityfiltered_reads`. If running with `fastp`, complexity filtering happens inclusively within the earlier shortread preprocessing step. Therefore there will not be an independent pipeline step for complexity filtering, and no independent FASTQ file (i.e. `--save_complexityfiltered_reads` will be ignored) - your complexity filtered reads will also be in the `fastp/` folder in the same file(s) as the preprocessed read.
 
+**We do not recommend performing any read preprocessing or complexity filtering if you are using ONTs Guppy toolkit for basecalling and post-processing.**
```
Comment (jfy133) [inline docs/usage.md:230]:
```suggestion
> ⚠️ For nanopore data: we do not recommend performing any read preprocessing or complexity filtering if you are using ONTs Guppy toolkit for basecalling and post-processing.
```

File: docs/usage.md [hunk 4]
```diff
@@ -475,6 +480,101 @@ We recommend adding the following line to your environment to limit this (typica
 NXF_OPTS='-Xms1g -Xmx4g'
 ```
 
+## Tutorials
+
+### Tutorial - How to create your custom database
+
```
Comment (jfy133) [inline docs/usage.md:486]:
```suggestion

Here we will give brief guidance on how to build databases for each supported taxonomic profiler. You should always consult the documentation of each tool for more information, how we provide these as quick reference guides.

The following tutorial assumes you already have the tool available (e.g. installed locally, or via conda, docker etc.), and you have already downloaded the FASTA files you wish to build into a database.

```

File: docs/usage.md [hunk 5]
```diff
@@ -475,6 +480,101 @@ We recommend adding the following line to your environment to limit this (typica
...
+--download-library archaea --db $DBNAME
+```
+
+To add more genomes:
+
+```bash
+kraken2-build --add-to-library genome.fa --db $DBNAME
+```
+
+You can follow Kraken2 [tutorial](https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown#custom-databases) for a more detailed description.
```
Comment (jfy133) [inline docs/usage.md:511]:
```suggestion

> These are instructions are based on Kraken 2.1.2

To build a Kraken2 database you need two components: a taxonomy (consisting of `names.dmp`, `nodes.dmp`, and `*accession2taxid`) files, and the FASTA files you wish to include.

To install pull the NCBI taxonomy you can run the following:

```bash
kraken2-build --download-taxonomy --db <YOUR_DB_NAME>
```

You can then add your FASTA files with the following build command

```bash
kraken2-build --add-to-library *.fna --db <YOUR_DB_NAME>
```

> 🛈 You can repeat this step multiple times to iteratively add more genomes prior building

You can also automatical download and add 'standard' libraries provided by Kraken2 (e.g. bacteria on RefSeq)

```bash
kraken2-build --download-library bacteria --db <YOUR_DB_NAME>
```

Once all genomes are added to the library, you can build the database (and optionally clean it up):

```bash
kraken2-build --build --db <YOUR_DB_NAME>
kraken2-build --clean--db <YOUR_DB_NAME>
```

You can then add the `<YOUR_DB_NAME>/` path to your nf-core/taxprofiler database input sheet.

You can follow the Kraken2 [tutorial](https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown#custom-databases) for a more detailed description.
```

File: docs/usage.md [hunk 6]
```diff
@@ -475,6 +480,101 @@ We recommend adding the following line to your environment to limit this (typica
...
+```
+
+#### MALT
+
+To create a custom database for MALT, the user should download and unzip the following database which lists all NCBI records. The input files are specified using -i and the index is specified using -d. A detailed description for each argument can be found [here](https://software-ab.informatik.uni-tuebingen.de/download/malt/manual.pdf)
+
+```bash
+wget https://software-ab.informatik.uni-tuebingen.de/download/megan6/megan-nucl-Feb2022.db.zip
+unzip megan-nucl-Feb2022.db
+malt-build -i path/to/fasta/files/*.{fna,fa} -s DNA -d index -t 8 -st 4 -a2t megan-nucl-Feb2022.db
```
Comment (jfy133) [inline docs/usage.md:546]:
```suggestion

MALT does not provide any default databases for profiling, therefore you must build your own.

You need FASTA files to include, and an (unzipped) [MEGAN mapping 'db' file](https://software-ab.informatik.uni-tuebingen.de/download/megan6/) for your FASTA type.

In addition to the input directory, output directory, and the mapping file database, you also need to specify the sequence type (DNA or Protein) with the `-s` flag.

```bash
malt-build -i <path>/<to>/<fasta>/*.{fna,fa,fasta} -a2t <path>/<to>/<map>.db -d <YOUR_DB_NAME>/  -s DNA
```

You can then add the `<YOUR_DB_NAME>/` path to your nf-core/taxprofiler database input sheet.

> ⚠️ MALT generates very large database files and requires large amounts of RAM. You can reduce both by increasing the step size `-st` (with a reduction in sensitivity).

> 🛈 MALT-build can be multi-threaded with `-t` to speed up building.

See the [MALT manual](https://software-ab.cs.uni-tuebingen.de/download/malt/welcome.html) for more information.

**Review by jfy133 — COMMENTED** [review]
@sofstam when I was reviewing the ones for the tools I assigned to me, I had a thought that describing it ina  slightly different way might be more useful, i.e. 1) what is the input, 2) what command do you need to run, 3) what goes into taxprofiler

See my suggestions and let me know what you think. If you agree, please commit the suggestions and I'll update the other tools that I'm assigned to in the same scheme.

**Comment by sofstam** [comment]
> @sofstam when I was reviewing the ones for the tools I assigned to me, I had a thought that describing it ina slightly different way might be more useful, i.e. 1) what is the input, 2) what command do you need to run, 3) what goes into taxprofiler
> 
> See my suggestions and let me know what you think. If you agree, please commit the suggestions and I'll update the other tools that I'm assigned to in the same scheme.

I have commited your suggestions, I think they are good.

File: docs/usage.md [hunk 7]
```diff
@@ -184,7 +185,7 @@ work                # Directory containing the nextflow working files
 
 ### Sequencing quality control
 
-nf-core taxprofiler offers [`falco`][https://github.com/smithlabcode/falco] as an alternative option to [`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/).
+[`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. nf-core taxprofiler offers [`falco`](https://github.com/smithlabcode/falco) as an alternative option.
```
Comment (jfy133) [inline docs/usage.md:188]:
```suggestion
[`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. nf-core taxprofiler offers [`falco`](https://github.com/smithlabcode/falco) as an drop-in replacement, with supposedly better improvement particularly for long reads.
```

File: docs/usage.md [hunk 8]
```diff
@@ -475,6 +480,121 @@ We recommend adding the following line to your environment to limit this (typica
...
+⚠️ MALT generates very large database files and requires large amounts of RAM. You can reduce both by increasing the step size `-st` (with a reduction in sensitivity).
+
+MALT-build can be multi-threaded with `-t` to speed up building.
+
+See the [MALT manual](https://software-ab.informatik.uni-tuebingen.de/download/malt/manual.pdf) for more information.
+
+#### Bracken
+
+You can follow Bracken [tutorial](https://ccb.jhu.edu/software/bracken/index.shtml?t=manual) to build a custom database. Alternatively, you can use one of the indexes that can be found [here](https://benlangmead.github.io/aws-indexes/k2).
+
```
Comment (jfy133) [inline docs/usage.md:572]:
```suggestion

Bracken does not provide any default databases for profiling, but rather building upon Kraken2 databases. See [Kraken2](#kraken2) for more information on how to build these.

In addition to a Kraken2 database, you also need to have the (average) read lengths (in bp) of your sequencing experiment, the K-mer size used to build the Kraken2 database, and Kraken2 available on your machine.

```bash
bracken-build -d <KRAKEN_DB_DIR> -k <KRAKEN_DB_KMER_LENGTH> -l <READLENGTH>
```

> 🛈 You can speed up database construction by supplying the threads parameter  (`-t`).

> 🛈 If you do not have Kraken2 in your `$PATH` you can point to the binary with `-x /<path>/<to>/kraken2`.

You can follow Bracken [tutorial](https://ccb.jhu.edu/software/bracken/index.shtml?t=manual) for more information. Alternatively, you can use one of the indexes that can be found [here](https://benlangmead.github.io/aws-indexes/k2).

```

File: docs/usage.md [hunk 9]
```diff
@@ -475,6 +438,234 @@ We recommend adding the following line to your environment to limit this (typica
 NXF_OPTS='-Xms1g -Xmx4g'
 ```
 
+## Tutorials
+
+### Tutorial - How to create your custom database
```
Comment (jfy133) [inline docs/usage.md:443]:
```suggestion
### Retrieving databases or building custom databases
```

**Review by jfy133 — COMMENTED** [review]
Adding MP3 and mOTUs

File: docs/usage.md [hunk 10]
```diff
@@ -475,6 +438,234 @@ We recommend adding the following line to your environment to limit this (typica
...
+⚠️ MALT generates very large database files and requires large amounts of RAM. You can reduce both by increasing the step size `-st` (with a reduction in sensitivity).
+
+MALT-build can be multi-threaded with `-t` to speed up building.
+
+See the [MALT manual](https://software-ab.informatik.uni-tuebingen.de/download/malt/manual.pdf) for more information.
+
+#### MetaPhlAn3
+
+<details markdown="1">
+<summary>Output files</summary>
```
[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md]: Implemented

File: docs/usage.md [hunk 11]
```diff
@@ -475,6 +438,261 @@ We recommend adding the following line to your environment to limit this (typica
...
+<details markdown="1">
+<summary>Expected files in database directory</summary>
+
+- `kaiju`
+  - `kaiju_db_*.fmi`
+  - `nodes.dmp`
+  - `names.dmp`
+
+</details>
+
```
Comment (jfy133) [inline docs/usage.md:543]: Missing what you need (i.e., what types of input files etc.

File: docs/usage.md [hunk 12]
```diff
@@ -475,6 +438,261 @@ We recommend adding the following line to your environment to limit this (typica
...
+It is possible to [create custom databases](https://github.com/bioinformatics-centre/kaiju#custom-database) with Kaiju.
+
+```bash
+kaiju-mkbwt -n 5 -a ACDEFGHIKLMNPQRSTVWY -o proteins proteins.faa
```
Comment (jfy133) [inline docs/usage.md:547]: what is `-n` and `-a`? Are they required?

Comment (jfy133) [reply docs/usage.md:547]: If `-n` is threads, see other tools how to add it (normlaly in a quote block)

**Review by jfy133 — COMMENTED** [review]
Some corrections where I can't make suggestions:

> It is unclear whether Bracken is suitable for running long reads, as it makes certain assumptions about read lengths. Furthemore, during testing we found issues where Bracken would fail on the long-read test data. Therefore nf-core/taxprofiler does not run Bracken on data specified as being sequenced with `OXFORD_NANOPORE` in the input samplesheet. If you believe this to be wrong, please contact us on the nf-core slack and we can discuss this.

If you believe this to be wrong -> If you would like to change this behaviour, <...>

> DIAMOND only allows output of a single format at a time, therefore parameters such --diamond_save_reads supplied will result in only aligned reads in SAM format will be produced, no taxonomic profiles will be available. Be aware of this when setting up your pipeline runs, depending n your particular use case.

`depending n your` should be `depending on your`

> Krona can only be run on MALT output if path to Krona taxonomy database supplied to `--krona_taxonomy_directory`. Therefore if you do not supply the a KRona directory, Krona plots will not be produced for MALT.

KRona -> Krona

File: docs/usage.md [hunk 13]
```diff
@@ -475,6 +438,261 @@ We recommend adding the following line to your environment to limit this (typica
...
+- `krakenuniq`
+  - `opts.k2d`
+  - `hash.k2d`
+  - `taxo.k2d`
+  - `database.idx`
+  - `taxDB`
+
+</details>
+
+For KrakenUniq, we recommend using one of the available databases [here](https://benlangmead.github.io/aws-indexes/k2). But if you wish to build your own, please see the [documentation](https://github.com/fbreitwieser/krakenuniq/blob/master/README.md#custom-databases-with-ncbi-taxonomy).
```
[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md:610]: Adressed in the upcoming PR

File: docs/usage.md [hunk 14]
```diff
@@ -475,6 +434,294 @@ We recommend adding the following line to your environment to limit this (typica
...
+MT192765.1  2697049
+```
+
+Then make a directory (`<DB_DIR_NAME>/`), containing the `seqid2taxid.map` file and your FASTA files in a subdirectory called `library/`. You must then run the `taxonomy` command on the `<DB_DIR_NAME>/` directory, and then build it.
+
+```
+mkdir -p <DB_DIR_NAME>/library
+mv `seqid2taxid.map` <DB_DIR_NAME>/
+mv *.fna  <DB_DIR_NAME>/library
+krakenuniq-build --db <DB_DIR_NAME> --kmer-len 31
```
Comment (jfy133) [inline docs/usage.md:620]:
```suggestion
krakenuniq-download --db <DB_DIR_NAME>  taxonomy
krakenuniq-build --db <DB_DIR_NAME> --kmer-len 31
```

Comment (jfy133) [reply docs/usage.md:620]: Missed a bit :grimacing:

File: docs/usage.md [hunk 15]
```diff
@@ -475,6 +438,261 @@ We recommend adding the following line to your environment to limit this (typica
...
+  - `taxo.k2d`
+  - `database.kraken`
+  - `database100mers.kmer_distrib`
+  - `database100mers.kraken`
+  - `database150mers.kmer_distrib`
+  - `database150mers.kraken`
+
+</details>
+
+Bracken does not provide any default databases for profiling, but rather building upon Kraken2 databases. See [Kraken2](#kraken2) for more information on how to build these.
```
[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md]: This is resolved

File: docs/usage.md (hunk 12 above)
Comment (sofstam) [reply docs/usage.md:533]: Same here

File: docs/usage.md [hunk 16]
```diff
@@ -220,10 +179,14 @@ Complexity filtering is primarily a run-time optimisation step. It is not necess
...
+There is one option for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong)
+
+The tools offer different algorithms and parameters for removing low complexity reads and quality filtering. We therefore recommend reviewing the pipeline's [parameter documentation](https://nf-co.re/taxprofiler/parameters) and the documentation of the tools (see links above) to decide on optimal methods and parameters for your dataset.
 
 You can optionally save the FASTQ output of the run merging with the `--save_complexityfiltered_reads`. If running with `fastp`, complexity filtering happens inclusively within the earlier shortread preprocessing step. Therefore there will not be an independent pipeline step for complexity filtering, and no independent FASTQ file (i.e. `--save_complexityfiltered_reads` will be ignored) - your complexity filtered reads will also be in the `fastp/` folder in the same file(s) as the preprocessed read.
 
+> ⚠️ For nanopore data: we do not recommend performing any read preprocessing or complexity filtering if you are using ONTs Guppy toolkit for basecalling and post-processing.
```
[the remark this answers is not in the document]
Comment (sofstam) [reply docs/usage.md:182]: Yes, exactly.

**Comment by sofstam** [comment]
nf-core-bot fix linting

**Review by jfy133 — APPROVED** [review]
Overall as a first draft LGTM, I will merge in and make up a follow up PR if I see any content missing when rendered on the website :+1:

**Pull request merged**